src/firefly/js/visualize/ui/CatalogSelectViewPanel.jsx

```diff
@@ -0,0 +1,623 @@
+import {dispatchTableSearch} from '../../tables/TablesCntlr.js';
+import {HelpIcon} from '../../ui/HelpIcon.jsx';
+import {CatalogSearchMethodType, SpatialMethod} from '../../ui/CatalogSearchMethodType.jsx';
+import {showInfoPopup} from '../../ui/PopupUtil.jsx';
+import {parseWorldPt} from '../../visualize/Point.js';
+import {VoSearchPanel} from '../../ui/VoSearchPanel.jsx';
+import {FileUpload} from '../../ui/FileUpload.jsx';
+import {convertAngle} from '../VisUtil';
+
+import './CatalogTableListField.css';
+
+/**
+ * group key for fieldgroup comp
+ */
+const gkey = 'CATALOG_PANEL';
+
+/**define the helpButton*/
+const helpIdStyle = {'textAlign': 'center', display: 'inline-block', height: 40, marginRight: 20};
+
+const dropdownName = 'IrsaCatalogDropDown';
+
+const initRadiusArcSec = parseFloat(500 / 3600).toString();
+
+/**
+ * Globally scoped here, master table, columns object
+ * @type {Array}
+ */
+var catmaster = [], cols = [];
+
+export class CatalogSelectViewPanel extends Component {
+
+    constructor(props) {
+        super(props);
+        this.state = {fields: FieldGroupUtils.getGroupFields(gkey)};
+    }
+
+    componentWillUnmount() {
+        this.iAmMounted = false;
+        if (this.unbinder) this.unbinder();
+    }
+
+    componentDidMount() {
+        this.iAmMounted = true;
+        this.unbinder = FieldGroupUtils.bindToStore(gkey, (fields) => {
+            if (fields !== this.state.fields && this.iAmMounted) {
+                this.setState({fields});
+            }
+        });
+    }
+
+    render() {
+        var {fields}= this.state;
+        return (
+            <div style={{padding: 10}}>
+                <FormPanel
+                    groupKey={gkey}
+                    onSubmit={(request) => onSearchSubmit(request)}
+                    onCancel={hideSearchPanel}>
+                    <CatalogSelectView fields={fields}/>
+                </FormPanel>
+            </div>
+        );
+    }
+}
+
+function onSearchSubmit(request) {
+    console.log('request ' + JSON.stringify(request));
+
+    if (request.Tabs === 'catalog') {
+        const wp = parseWorldPt(request[ServerParams.USER_TARGET_WORLD_PT]);
+        if (!wp && (request.spatial === SpatialMethod.Cone.value
+            || request.spatial === SpatialMethod.Box.value
+            || request.spatial === SpatialMethod.Elliptical.value)) {
+            showInfoPopup('Target is required');
+            return;
+        }
+        doCatalog(request);
+    }
+    else if (request.Tabs === 'loadcat') {
+        doLoadTable(request);
+    }
+    else if (request.Tabs === 'vosearch') {
+        doVoSearch(request);
+    }
+    else {
+        console.log('request no supported');
+    }
+}
+function hideSearchPanel() {
+    dispatchHideDropDown();
+}
+
+function doCatalog(request) {
+
+    const conesize = convertAngle('deg', 'arcsec', request.conesize);
+    var title = request.project + '-' + request.cattable + ' (' + request.spatial;
```

Comment:
good point, i'll change it to use template string instead.

On 5/14/16 11:03 PM, Trey Roby wrote:
> |var title =` ${request.project}-${request.cattable}(${request.spatial}`|

-- 
Emmanuel Joliet
NASA/IPAC  Infrared Science Archive
MS 100-22, Caltech, Pasadena, CA 91125
Office: 170
Email:  ejoliet@ipac.caltech.edu
Phone:  (626) 395-1489